hs/app/reach-cli/Main.hs

```diff
@@ -0,0 +1,352 @@
+{-# OPTIONS_GHC -fno-warn-type-defaults #-}
+
+{-# LANGUAGE ExtendedDefaultRules #-}
+{-# LANGUAGE OverloadedStrings    #-}
+
+module Main (main) where
+
+import Control.Monad.Shell
+import Options.Applicative
+import Options.Applicative.Help.Pretty
+import System.Posix.IO (stdError)
+
+import qualified Data.Text.Lazy    as T
+import qualified Data.Text.Lazy.IO as T
+
+default (T.Text)
+
+-- TODO usage help currently prints `reach-cli` instead of `reach`, which may
+-- be confusing to end-users who shouldn't have to care that their CLI session
+-- is being forwarded to this binary... Consider renaming
+
+-- TODO
+reachVersion :: T.Text
+reachVersion = "0.1"
+
+reachImages :: [T.Text]
+reachImages =
+  [ "reach"
+  , "ethereum-devnet"
+  , "algorand-devnet"
+  , "devnet-cfx"
+  , "runner"
+  , "react-runner"
+  , "rpc-server"
+  ]
+
+
+data Cli
+  = Clean FilePath T.Text
+  | Compile
+  | Devnet
+  | DockerReset
+  | Down
+  | Hashes
+  | Help
+  | Init
+  | NumericVersion
+  | React
+  | ReactDown
+  | RpcRun
+  | RpcServer
+  | RpcServerDown
+  | Run
+  | Scaffold
+  | Unscaffold
+  | Update
+  | Upgrade
+  | Version
+  | Whoami
+
+
+--------------------------------------------------------------------------------
+cmdClean :: Mod CommandFields Cli
+cmdClean = command "clean" . info opts $ fullDesc <> desc <> fdoc
+ where
+  desc = progDesc "Delete 'build/$MODULE.$IDENT.mjs'"
+  fdoc = footerDoc . Just
+     $  text "MODULE is \"index\" by default"
+   <$$> text "IDENT  is \"main\"  by default"
+   <$$> text ""
+   <$$> text "If:"
+   <$$> text " * MODULE is a directory then `cd $MODULE && rm -f \"build/index.$IDENT.mjs\";"
+   <$$> text " * MODULE is <something-else> then `rm -f \"build/$MODULE.$IDENT.mjs\""
+  opts = Clean
+    <$> strArgument (metavar "MODULE" <> value "index" <> showDefault)
+    <*> strArgument (metavar "IDENT"  <> value "main"  <> showDefault)
+
+
+-- | > rm -f "build/$m.$i.mjs
+clean :: FilePath -> T.Text -> Script ()
```

Comment:
Can there not be 5x triplicate duplication of information?

`cmdClean`, `clean`, `Clean`, `cmds`, and `main` all have the same information